src/Host/Broker/Impl/Security/SecurityManager.cs

```diff
@@ -48,7 +52,29 @@ public class SecurityManager {
             return new ClaimsPrincipal(identity);
         }
 
-        private ClaimsPrincipal SignInUsingLogon(BasicSignInContext context) {
+        private async Task<RUserProfileCreateResponse> CreateProfileAsync(RUserProfileCreateRequest request, CancellationToken ct) {
+            using (NamedPipeClientStream client = new NamedPipeClientStream("Microsoft.R.Host.UserProfile.Creator{b101cc2d-156e-472e-8d98-b9d999a93c7a}")) { 
+                try {
+                    await client.ConnectAsync(ct);
+
+                    string jsonReq = JsonConvert.SerializeObject(request);
+                    byte[] data = Encoding.Unicode.GetBytes(jsonReq.ToString());
+
+                    await client.WriteAsync(data, 0, data.Length, ct);
+                    await client.FlushAsync(ct);
+
+                    byte[] responseRaw = new byte[1024];
+                    var bytesRead = await client.ReadAsync(responseRaw, 0, responseRaw.Length, ct);
+                    string jsonResp = Encoding.Unicode.GetString(responseRaw, 0, bytesRead);
+                    return JsonConvert.DeserializeObject<RUserProfileCreateResponse>(jsonResp);
+                } catch (Exception ex) when (!ex.IsCriticalException()) {
+                    _logger.LogError(Resources.Error_ProfileCreationFailedIO, request.Username);
+                    return RUserProfileCreateResponse.Blank;
+                }
+            }
+        }
+
+        private async Task<ClaimsPrincipal> SignInUsingLogonAsync(BasicSignInContext context) {
```

Comment:
If I remember correctly, it's an external type from the package that we use for HTTP Basic. Out of the box, it doesn't do secure strings.

We can fork it and make our own implementation, but it would still need to make a secure string out of a plain string that it gets from HTTP header dictionary, which makes it moot to try to conceal it.

This is probably best left for a separate PR that would deal with this problem end to end (i.e. switch to some other auth scheme that does not have password unencrypted at any point, not even in the headers).